core/src/jsMain/kotlin/Bluetooth.kt

```diff
@@ -1,62 +1,36 @@
 package com.juul.kable
 
-import com.juul.kable.Options.Filter.Name
-import com.juul.kable.Options.Filter.NamePrefix
-import com.juul.kable.Options.Filter.Services
-import com.juul.kable.external.Bluetooth
-import kotlinx.coroutines.CoroutineScope
-import kotlin.js.Promise
+import com.juul.kable.Bluetooth.Availability.Available
+import com.juul.kable.Bluetooth.Availability.Unavailable
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import com.juul.kable.external.Bluetooth as JsBluetooth
+
+public actual enum class Reason
+
+private const val AVAILABILITY_CHANGED = "availabilitychanged"
+
+internal actual val bluetoothAvailability: Flow<Bluetooth.Availability> = callbackFlow {
+    // https://developer.mozilla.org/en-US/docs/Web/API/Bluetooth/onavailabilitychanged
+    val listener: (dynamic) -> Unit = { event ->
+        val isAvailable = event.value as Boolean
+        trySend(if (isAvailable) Available else Unavailable(reason = null))
+    }
+
+    bluetooth.apply {
+        addEventListener(AVAILABILITY_CHANGED, listener)
+        awaitClose {
+            removeEventListener(AVAILABILITY_CHANGED, listener)
+        }
+    }
+}
```

Comment:
It looks like this will crash on the `checkNotNull` below if there is no bluetooth object, is that preferred to emitting `Unavailable`?

Also, if the browser does not support the `availabilitychanged` event this will never emit.

Perhaps it would be useful to emit `Unavailable` with a suitable reason for both these scenarios instead?